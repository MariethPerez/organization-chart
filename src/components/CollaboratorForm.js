import React from "react";
import styled from "@emotion/styled";

function CollaboratorForm({ addNewChild, setOpenedForm }) {
  const CollaboratorForm = styled.div`
    display: flex;
    align-items: center;
    position: fixed;
    z-index: 1;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    overflow: auto;
    background-color: rgba(0, 0, 0, 0.4);
  `;

  const Header = styled.h1`
    text-align: center;
    margin: 15px;
    font-size: 40px;
    color: white;
    font-weight: bolder;
  `;

  const FormInput = styled.form`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 500px;
    padding: 1.5em;
    background: black;
    border-radius: 0.5em;
    margin: 0 auto;
  `;

  const InputName = styled.input`
    margin: 1em 0;
    font-size: 30px;
    padding-top: 10px;
    padding-bottom: 10px;
    border-radius: 4px;
    outline: none;
    border: none;
    border-bottom: 1px solid #ddd;
    width: 100%;
    text-align: center;
    &:focus {
      border: 1px solid red;
    }
  `;

  const SubmitName = styled.input`
    line-height: 1.5em;
    background: darkgray;
    border: none;
    color: white;
    font-size: 20px;
    font-weight: bold;
    padding: 0.5em;
    border-radius: 0.25rem;
    text-align: center;
    width: 50%;
    cursor: pointer;
    &:hover {
      background-color: #484848;
    }
  `;

  const Cancel = styled.span`
    font-size: 2em;
    cursor: pointer;
    color: white;
    text-align: right;
    width: 100%;
  `;

  function handleSubmit(event) {
    event.preventDefault();
    addNewChild(event.target.elements.name.value);
    setOpenedForm(false);
  }

  function handleCancel(event) {
    setOpenedForm(false);
  }

  return (
    <CollaboratorForm>
      <FormInput onSubmit={handleSubmit}>
        <Cancel onClick={handleCancel}>&times;</Cancel>
        <Header>Collaborator name</Header>
        <label name="name" aria-label="Write Your Name">
          <InputName
            aria-label="Get Collaborator Name"
            id="my_name"
            name="name"
            type="text"
            placeholder="Enter colaborator name"
          />
        </label>
        <SubmitName type="submit" value="submit" />
      </FormInput>
    </CollaboratorForm>
  );
}

export default CollaboratorForm;
