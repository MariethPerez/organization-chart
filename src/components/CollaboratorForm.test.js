import React from "react";
import { render, fireEvent, waitForElement } from "@testing-library/react";

import CollaboratorForm from "./CollaboratorForm";

test("CollaboratorForm", () => {
  const addNewChild = jest.fn();
  const setOpenedForm = jest.fn();
  const { asFragment } = render(
    <CollaboratorForm addNewChild={addNewChild} setOpenedForm={setOpenedForm} />
  );

  expect(asFragment()).toMatchSnapshot();
});

test("CreateCollaborator", async () => {
  const addNewChild = jest.fn();
  const setOpenedForm = jest.fn();
  const { getByLabelText } = render(
    <CollaboratorForm addNewChild={addNewChild} setOpenedForm={setOpenedForm} />
  );

  const input = await waitForElement(() =>
    getByLabelText("Get Collaborator Name")
  );
  fireEvent.change(input, {
    target: {
      value: "ColaboratorJuan"
    }
  });
  fireEvent.submit(input);

  expect(addNewChild).toHaveBeenCalledWith("ColaboratorJuan");
});
