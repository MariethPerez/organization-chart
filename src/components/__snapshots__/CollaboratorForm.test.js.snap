// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollaboratorForm 1`] = `
<DocumentFragment>
  .emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  z-index: 2;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0,0,0,0.4);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 500px;
  padding: 1.5em;
  background: black;
  border-radius: 0.5em;
  margin: 0 auto;
}

.emotion-0 {
  font-size: 2em;
  cursor: pointer;
  color: white;
  text-align: right;
  width: 100%;
}

.emotion-1 {
  text-align: center;
  margin: 15px;
  font-size: 40px;
  color: white;
  font-weight: bolder;
}

.emotion-2 {
  margin: 1em 0;
  font-size: 30px;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 4px;
  outline: none;
  border: none;
  border-bottom: 1px solid #ddd;
  width: 100%;
  text-align: center;
}

.emotion-2:focus {
  border: 1px solid red;
}

.emotion-3 {
  line-height: 1.5em;
  background: darkgray;
  border: none;
  color: white;
  font-size: 20px;
  font-weight: bold;
  padding: 0.5em;
  border-radius: 0.25rem;
  text-align: center;
  width: 50%;
  cursor: pointer;
}

.emotion-3:hover {
  background-color: #484848;
}

<div
    class="emotion-5"
  >
    <form
      class="emotion-4"
    >
      <span
        aria-label="Close create collaborator"
        class="emotion-0"
      >
        ×
      </span>
      <h1
        class="emotion-1"
      >
        Collaborator name
      </h1>
      <label
        aria-label="Write Your Name"
        name="name"
      >
        <input
          aria-label="Get Collaborator Name"
          class="emotion-2"
          id="my_name"
          name="name"
          placeholder="Enter colaborator name"
          type="text"
        />
      </label>
      <input
        class="emotion-3"
        type="submit"
        value="submit"
      />
    </form>
  </div>
</DocumentFragment>
`;

exports[`CreateCollaborator UI 1`] = `
<DocumentFragment>
  .emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  z-index: 2;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0,0,0,0.4);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 500px;
  padding: 1.5em;
  background: black;
  border-radius: 0.5em;
  margin: 0 auto;
}

.emotion-0 {
  font-size: 2em;
  cursor: pointer;
  color: white;
  text-align: right;
  width: 100%;
}

.emotion-1 {
  text-align: center;
  margin: 15px;
  font-size: 40px;
  color: white;
  font-weight: bolder;
}

.emotion-2 {
  margin: 1em 0;
  font-size: 30px;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 4px;
  outline: none;
  border: none;
  border-bottom: 1px solid #ddd;
  width: 100%;
  text-align: center;
}

.emotion-2:focus {
  border: 1px solid red;
}

.emotion-3 {
  line-height: 1.5em;
  background: darkgray;
  border: none;
  color: white;
  font-size: 20px;
  font-weight: bold;
  padding: 0.5em;
  border-radius: 0.25rem;
  text-align: center;
  width: 50%;
  cursor: pointer;
}

.emotion-3:hover {
  background-color: #484848;
}

<div
    class="emotion-5"
  >
    <form
      class="emotion-4"
    >
      <span
        aria-label="Close create collaborator"
        class="emotion-0"
      >
        ×
      </span>
      <h1
        class="emotion-1"
      >
        Collaborator name
      </h1>
      <label
        aria-label="Write Your Name"
        name="name"
      >
        <input
          aria-label="Get Collaborator Name"
          class="emotion-2"
          id="my_name"
          name="name"
          placeholder="Enter colaborator name"
          type="text"
        />
      </label>
      <input
        class="emotion-3"
        type="submit"
        value="submit"
      />
    </form>
  </div>
</DocumentFragment>
`;

exports[`CreateCollaborator UI 2`] = `
<DocumentFragment>
  .emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  z-index: 2;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0,0,0,0.4);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 500px;
  padding: 1.5em;
  background: black;
  border-radius: 0.5em;
  margin: 0 auto;
}

.emotion-0 {
  font-size: 2em;
  cursor: pointer;
  color: white;
  text-align: right;
  width: 100%;
}

.emotion-1 {
  text-align: center;
  margin: 15px;
  font-size: 40px;
  color: white;
  font-weight: bolder;
}

.emotion-2 {
  margin: 1em 0;
  font-size: 30px;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 4px;
  outline: none;
  border: none;
  border-bottom: 1px solid #ddd;
  width: 100%;
  text-align: center;
}

.emotion-2:focus {
  border: 1px solid red;
}

.emotion-3 {
  line-height: 1.5em;
  background: darkgray;
  border: none;
  color: white;
  font-size: 20px;
  font-weight: bold;
  padding: 0.5em;
  border-radius: 0.25rem;
  text-align: center;
  width: 50%;
  cursor: pointer;
}

.emotion-3:hover {
  background-color: #484848;
}

<div
    class="emotion-5"
  >
    <form
      class="emotion-4"
    >
      <span
        aria-label="Close create collaborator"
        class="emotion-0"
      >
        ×
      </span>
      <h1
        class="emotion-1"
      >
        Collaborator name
      </h1>
      <label
        aria-label="Write Your Name"
        name="name"
      >
        <input
          aria-label="Get Collaborator Name"
          class="emotion-2"
          id="my_name"
          name="name"
          placeholder="Enter colaborator name"
          type="text"
        />
      </label>
      <input
        class="emotion-3"
        type="submit"
        value="submit"
      />
    </form>
  </div>
</DocumentFragment>
`;
